lms/djangoapps/course_blocks/transformers/tests/helpers.py

```diff
@@ -115,7 +116,10 @@ def add_parents(self, block_hierarchy, block_map):
             # It would be re-added to the course if the course was
             # explicitly listed in parents.
             course = modulestore().get_item(block_map['course'].location)
-            course.children.remove(block_key)
+            try:
+                course.children.remove(block_key)
+            except ValueError:
+                pass  # block_key was not a direct child of course, carry on
```

Comment:
It is still needed.  What's happening is that the code expects that blocks are either defined at the top level and have explicit `#parents` defined, or are included as the children of some other node.  But the docs for `add_parent` suggest that you should be able to create a node in the `#children` of another node, but define extra parents with `#parents`.  But those nodes won't have the course listed as their parent, so the original version of this blows up.  I think it's a use case that the author of this code intended to support (evidenced by the docstring), but it never got used, so the kinks didn't worked out.  I think we should keep this, but maybe rewrite it as an `if`-condition instead of as a `try/except`.